src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1087,68 +1087,45 @@ internal virtual void FillRequestStream(HttpRequestMessage request)
 
             if (Form is not null)
             {
-                // Content headers will be set by MultipartFormDataContent which will throw unless we clear them first
-                WebSession.ContentHeaders.Clear();
-
                 var formData = new MultipartFormDataContent();
                 foreach (DictionaryEntry formEntry in Form)
                 {
                     // AddMultipartContent will handle PSObject unwrapping, Object type determination and enumerateing top level IEnumerables.
                     AddMultipartContent(fieldName: formEntry.Key, fieldValue: formEntry.Value, formData: formData, enumerate: true);
                 }
 
-                SetRequestContent(request, formData);
+                request.Content = SetRequestContent(request, formData);
```

Comment:
I wonder if now we don't return unused length why do we need to return Content? We pass `request` as argument so we can do the assignment in `SetRequestContent` internally as the method name says.

Reply:
`SetRequestContent` has to return the Content to use the switch expression

Reply:
Using switch expression is not mandatory 😺 This only force us to do many unneeded changes.